src/dispatch/builder.rs

```diff
@@ -253,13 +258,51 @@ impl<'a, 'b> DispatcherBuilder<'a, 'b> {
         writes.dedup();
 
         let accessor = BatchAccessor::new(reads, writes);
-        let dispatcher = dispatcher_builder.build();
+        let dispatcher: Dispatcher<'a, 'b> = dispatcher_builder.build();
 
-        let batch_system = unsafe { T::create(accessor, dispatcher) };
+        let batch_system = unsafe {
+            BatchControllerWrapper::<'a, 'b, T>::create(accessor, controller, dispatcher)
+        };
 
         self.add(batch_system, name, dep);
     }
 
+    pub fn with_multi_batch<T>(
```

Comment:
Re:

> The multi-batch is a separate method of the builder. Alternatively, it could be exposed as the wrapper type and one could create it directly. I'm not sure on what's better.

I think if we do this:

* Rename `MultiDispatchControllerImpl` to `MultiDispatchBatchController`
* Make that `pub`

Then the difference would be:

```rust
// from this
let mut dispatcher = DispatcherBuilder::new()
    .with_batch(
        Run3Times,
        // ..
    )

// to this
use shred::MultiDispatchController;
let mut dispatcher = DispatcherBuilder::new()
    .with_batch(
        MultiDispatchController::new(Run3Times),
        // ..
    )
```

Then we can remove `add_multi_batch` and `with_multi_batch`, so all batch controllers would be passed through the `with_batch` method.

That is, I think requiring people to import another pre-made type is okay

Reply:
I've went with `MultiDispatcher`. That `MultiDispatchBatchController` seems really long and `MultiDispatchController` is already taken by the trait.